File: mathGeometry/263_ugly_number.go
// O(log(n)) time | O(1) space 
func isUgly(n int) bool {
    if n <= 0 {
        return false
    }
    
    primeFactors := []int{2, 3, 5}
    for _, factor := range primeFactors {
        for n % factor == 0 {
            n /= factor
        }
    }
    return n == 1
}

File: mathGeometry/263_ugly_number.py
class Solution:
    # O(log(n)) time | O(1) space
    def isUgly(self, n: int) -> bool:
        if n <= 0: return False
        
        primeFactors = [2, 3, 5]
        for factor in primeFactors:
            while n % factor == 0:
                n /= factor
            
        return n == 1

